edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/AuthAgentController.cs

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Hub.Mqtt;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class AuthAgentController : Controller
+    {
+        const int Authenticated = 200;
+        const int Unauthenticated = 403;
+
+        static readonly string apiVersion = "2020-04-20";
+
+        readonly IAuthenticator authenticator;
+        readonly IUsernameParser usernameParser;
+        readonly IClientCredentialsFactory clientCredentialsFactory;
+        readonly AuthAgentProtocolHeadConfig config;
+
+        public AuthAgentController(IAuthenticator authenticator, IUsernameParser usernameParser, IClientCredentialsFactory clientCredentialsFactory, AuthAgentProtocolHeadConfig config)
+        {
+            this.authenticator = Preconditions.CheckNotNull(authenticator, nameof(authenticator));
+            this.usernameParser = Preconditions.CheckNotNull(usernameParser, nameof(usernameParser));
+            this.clientCredentialsFactory = Preconditions.CheckNotNull(clientCredentialsFactory, nameof(clientCredentialsFactory));
+            this.config = Preconditions.CheckNotNull(config, nameof(config));
+        }
+
+        [HttpPost]
+        [Produces("application/json")]
+        public async Task<JsonResult> HandleAsync([FromBody] AuthRequest request)
+        {
+            if (request == null)
+            {
+                Events.ErrorDecodingPayload();
+                return this.Json(GetErrorResult());
+            }
+
+            if (!string.Equals(request.Version, apiVersion))
+            {
+                Events.ErrorBadVersion(request.Version ?? "(none)");
+                return this.Json(GetErrorResult());
+            }
+
+            try
+            {
+                var isAuthenticated = default(bool);
```

Comment:
Nit: Why do you need to initialize here?

Reply:
that is just to enable using 'var', so the variable names are aligned. also I didn't want to move the types into the deconstruction below, so it is easier to see what comes back. I thought it more readable because the variable names are not obscured by type definition, but the type is still there is one is interested